_posts/2018-11-02-optimising-microservices-part-1.md

```diff
@@ -0,0 +1,120 @@
+---
+layout:     post
+title:      Optimising an AWS microservice - Part 1
+date:       2018-11-02
+summary:    In this two part post we'll talk about our learnings from optimising an AWS microservice to reduce latency and hit non-functional targets
+author:     jamie_munro
+image:      optimising-aws-microservice/01-initial-system.png
+tags:       aws, serverless, cloud, microservices
+category:   Cloud Computing
+---
+
+In the Trading Models squad we build systems that calculate the probabilities of things happening in sporting events, and we use these probabilities to set Sky Bet’s prices. 
+Recently, we’ve been building out a pricing service in AWS for Sky Bet’s RequestABet product - an API that our internal sportsbook platform can use to get the price of football betting opportunities as customers request them. One thing that’s different about this project for our team is that the traffic through our system will be customer driven rather than trader driven meaning that we need to serve requests fast, at large scale.  
+In this two part post, I’ll take you through some of the optimisations we’ve made to our system to hit our non-functional requirements.
+
+## Our System
+![Initial System Diagram](/images/optimising-aws-microservice/01-initial-system.png)
+
+The above diagram shows our initial system design. There are three main components here:
+
+### The ‘Opinion’ Store 
+* ‘Opinion’ is our term for the trader inputs required for our model, i.e. a numerical representation of what we expect to happen in the game. It comprises things like the average number of goals we expect a team to score
+* This is stored in a DynamoDB table fronted by an API that allows the traders to PUT opinion and our system to GET it when required
+
+### The Pricing Model
+* This is a java lambda - you pass it a description of what might happen in a football match along with the necessary opinion and it returns a price.
+* E.G. you might request “Leeds Utd to score 3+ goals” and it might tell you this is 2/1
```

Comment:
It's "e.g.", but because it's at the start of a sentence, it should be "E.g." ;-) I'd just opt for spelling it out though:

```suggestion
* For example, you might request “Leeds Utd to score 3+ goals” and it might tell you this is 2/1
```